packages/pytest-simcore/src/pytest_simcore/minio_service.py

```diff
@@ -32,49 +44,53 @@ def minio_config(docker_stack: Dict, devel_environ: Dict) -> Dict[str, str]:
     }
 
     # nodeports takes its configuration from env variables
-    old_environ = deepcopy(os.environ)
     for key, value in config["client"].items():
-        os.environ[f"S3_{key.upper()}"] = str(value)
-    os.environ["S3_SECURE"] = devel_environ["S3_SECURE"]
-    os.environ["S3_BUCKET_NAME"] = config["bucket_name"]
+        monkeypatch_module.setenv(f"S3_{key.upper()}", str(value))
+
+    monkeypatch_module.setenv("S3_SECURE", devel_environ["S3_SECURE"])
+    monkeypatch_module.setenv("S3_BUCKET_NAME", config["bucket_name"])
 
-    yield config
-    # restore environ
-    os.environ = old_environ
+    return config
 
 
 @pytest.fixture(scope="module")
-def minio_service(minio_config: Dict[str, str]) -> S3Client:
-    assert wait_till_minio_responsive(minio_config)
+def minio_service(minio_config: Dict[str, str]) -> Iterator[Minio]:
+
+    client = Minio(**minio_config["client"])
+
+    for attempt in Retrying(
+        wait=tenacity.wait_fixed(5),
+        stop=tenacity.stop_after_attempt(60),
+        before_sleep=tenacity.before_sleep_log(log, logging.WARNING),
+        reraise=True,

```

Comment:
question: what is the difference with using the tenacity decorator already?

Reply:
effectively none. I just wanted to have everything in the same function because the ``wait_for_response`` is only used there. 
In any case, in PR #2281 I plan to have a collection of ``is_responsive``-like functions for all services ... including minio.